### PR TITLE
Fix merge conflict and update makefiles

### PR DESCRIPTION
**Pull request by keithc-ca**
* fix conflict in `make/autoconf/jdk-version.m4`
* adapt to removal of hook in `CompileJavaModules.gmk`

### CODE REVIEW

**Comment by keithc-ca** [comment]
Jenkins compile zlinux jdknext

**Comment by JasonFengJ9** [comment]
Is there a update to `openjdk-tag.gmk`?

**Comment by keithc-ca** [comment]
> Is there a update to `openjdk-tag.gmk`?

No, the latest tag is still 'jdk-17+4'.

File: closed/custom/modules/jdk.jcmd/Java.gmk [hunk 1]
```diff
@@ -1,11 +1,11 @@
 # ===========================================================================
-# (c) Copyright IBM Corp. 2017, 2020 All Rights Reserved
+# (c) Copyright IBM Corp. 2021, 2021 All Rights Reserved
 # ===========================================================================
 # This code is free software; you can redistribute it and/or modify it
 # under the terms of the GNU General Public License version 2 only, as
 # published by the Free Software Foundation.
 #
-# IBM designates this particular file as subject to the "Classpath" exception
+# IBM designates this particular file as subject to the "Classpath" exception
```
Comment (JasonFengJ9) [inline closed/custom/modules/jdk.jcmd/Java.gmk:8]: nitpicking: an extra space here

Comment (keithc-ca) [reply closed/custom/modules/jdk.jcmd/Java.gmk:8]: Thanks, I'll fix that.

Comment (keithc-ca) [reply closed/custom/modules/jdk.jcmd/Java.gmk]: Fixed.

**Comment by keithc-ca** [comment]
Successful build https://ci.eclipse.org/openj9/job/Build_JDKnext_s390x_linux_Personal/124/ should not be invalidated by the removal of some trailing whitespace.

File: closed/custom/modules/java.base/Copy.gmk [hunk 2]
```diff
@@ -20,6 +20,8 @@
 
 include $(TOPDIR)/closed/CopySupport.gmk
 
+COPY += ExternalMessages.properties
```
Comment (JasonFengJ9) [inline closed/custom/modules/java.base/Copy.gmk:23]: Copyright update missed in this file.

Comment (keithc-ca) [reply closed/custom/modules/java.base/Copy.gmk:23]: Fixed that too.

File: make/autoconf/jdk-options.m4 [hunk 3]
```diff
@@ -24,7 +24,7 @@
 #
 
 # ===========================================================================
-# (c) Copyright IBM Corp. 2020, 2020 All Rights Reserved
+# (c) Copyright IBM Corp. 2020, 2021 All Rights Reserved
```
Comment (JasonFengJ9) [inline make/autoconf/jdk-options.m4:27]: Is there an IBM specific change to this file in this PR, why change the Copyright?

Comment (keithc-ca) [reply make/autoconf/jdk-options.m4:27]: Yes, there still are changes relative to openjdk.

**Comment by pshipton** [comment]
Besides fixing the merge conflict, I don't think the changes should be buried in "Merge master HEAD into openj9-staging" but be a separate commit.

**Comment by keithc-ca** [comment]
The merge commit only addresses the merge conflict.
The other commit, 'Replace hooks in CompileJavaModules.gmk' includes the other changes.

File: make/autoconf/jdk-options.m4 (hunk 3 above)
Comment (JasonFengJ9) [reply make/autoconf/jdk-options.m4:27]: The only change made to this `jdk-options.m4` file was:
from
```
    # Use the date from version-numbers
```
to
```
    # Use the date from version-numbers.conf
```
This is not IBM specific, do I miss something?

**Comment by pshipton** [comment]
Neither ExternalMessages.properties or ibm_gpu_thresholds.properties are in the modules file in the PR build.

**Comment by keithc-ca** [comment]
> Neither `ExternalMessages.properties` or `ibm_gpu_thresholds.properties` are in the modules file in the PR build.

I thought I checked the `jmod` files and saw them; I'll investigate.

**Comment by keithc-ca** [comment]
Those two properties files now appear in the `jmod` files and the `modules` file as expected.

**Comment by pshipton** [comment]
Jenkins compile zlinux jdknext

File: make/autoconf/jdk-options.m4 (hunk 3 above)
Comment (pshipton) [reply make/autoconf/jdk-options.m4:27]: It won't stop me merging, but it does seem wrong that the only change to this file in the "Replace hooks in CompileJavaModules.gmk" commit is the copyright update. The IBM copyright shouldn't need to be updated due to OpenJDK changes. The IBM copyright won't get updated for everything that merges cleanly.

**Review by pshipton — APPROVED** [review]

File: make/autoconf/jdk-options.m4 (hunk 3 above)
Comment (keithc-ca) [reply make/autoconf/jdk-options.m4:27]: That file was changed in the merge commit, but I forgot to update the copyright date at that time.

**Pull request merged**